stdlib/3/optparse.pyi

```diff
@@ -0,0 +1,248 @@
+    def get_opt_string(self) -> str: ...
+    def process(self, opt, value, values: Any, parser: OptionParser) -> int: ...
+    def take_action(self, action, dest, opt, value, values, parser: OptionParser) -> int: ...
+    def takes_value(self) -> bool: ...
+
+make_option = Option
+
+class OptionContainer:
+    _long_opt = ...  # type: Dict[str, Any]
+    _short_opt = ...  # type: Dict[str, Any]
+    conflict_handler = ...  # type: Any
+    defaults = ...  # type: Dict[str, Any]
+    description = ...  # type: Any
+    option_class = ...  # type: Any
+    def __init__(self, option_class, conflict_handler, description) -> None: ...
+    def _check_conflict(self, option: Any) -> None: ...
+    def _create_option_mappings(self) -> None: ...
+    def _share_option_mappings(self, parser) -> None: ...
+    def add_option(self, *args, **kwargs) -> Any: ...
+    def add_options(self, option_list) -> None: ...
+    def destroy(self) -> None: ...
+    def format_description(self, formatter: Optional[HelpFormatter]) -> Any: ...
+    def format_help(self, formatter: HelpFormatter) -> str: ...
+    def format_option_help(self, formatter: Optional[HelpFormatter]) -> str: ...
+    def get_description(self) -> Any: ...
+    def get_option(self, opt_str) -> Optional[Option]: ...
+    def has_option(self, opt_str) -> bool: ...
+    def remove_option(self, opt_str) -> None: ...
+    def set_conflict_handler(self, handler) -> None: ...
+    def set_description(self, description) -> None: ...
+
+class OptionGroup(OptionContainer):
+    _long_opt = ...  # type: Dict[Any, Any]
+    _short_opt = ...  # type: Dict[Any, Any]
+    conflict_handler = ...  # type: Any
+    defaults = ...  # type: Dict[str, Any]
+    description = ...  # type: Any
+    option_class = ...  # type: Any
+    option_list = ...  # type: List
+    parser = ...  # type: Any
+    title = ...  # type: Any
+    def __init__(self, parser, title, *args, **kwargs) -> None: ...
+    def _create_option_list(self) -> None: ...
+    def format_help(self, formatter: HelpFormatter) -> Any: ...
+    def set_title(self, title) -> None: ...
+
+class OptionParser(OptionContainer):
+    _long_opt = ...  # type: Dict[str, Any]
+    _short_opt = ...  # type: Dict[Any, Any]
+    allow_interspersed_args = ...  # type: bool
+    conflict_handler = ...  # type: Any
+    defaults = ...  # type: Dict[Any, Any]
+    description = ...  # type: str
+    epilog = ...  # type: Any
+    formatter = ...  # type: HelpFormatter
+    largs = ...  # type: Union[None, List[str]]
+    option_class = ...  # type: Callable
+    option_groups = ...  # type: List[OptionParser]
+    option_list = ...  # type: List[Any]
+    process_default_values = ...  # type: Any
+    prog = ...  # type: Any
+    rargs = ...  # type: Optional[List[Any]]
+    standard_option_list = ...  # type: List
+    usage = ...  # type: Optional[str]
+    values = ...  # type: Any
+    version = ...  # type: str
+    def __init__(self, *args, **kwargs) -> None: ...
```

Comment:
Also needs more specific types. This is pretty important because this is one of the main entry points to the module that users are going to call.